File: app/src/main/java/com/tournamentbracketjava/Adapters/RowsAdapter.java
package com.tournamentbracketjava.Adapters;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import com.tournamentbracketjava.Models.Annotations;
import com.tournamentbracketjava.Models.Rows;
import com.tournamentbracketjava.R;

import java.util.List;

public class RowsAdapter extends RecyclerView.Adapter<RowsAdapter.MyHolder>{

    private List<Rows> rows;
    private List<Annotations> annotations;
    private int cardWidth,Width;
    private Context context;

    public RowsAdapter(List<Rows> rows, List<Annotations> annotations, int cardWidth, int Width, Context context) {
        this.rows = rows;
        this.annotations = annotations;
        this.cardWidth = cardWidth;
        this.Width = Width;
        this.context = context;
    }

    @NonNull
    @Override
    public MyHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(context).inflate(R.layout.single_row_card,parent,false);
        return new MyHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final MyHolder holder, @SuppressLint("RecyclerView") final int position) {

        holder.rowElemetsRecycle.setNestedScrollingEnabled(false);

        RowElemetsAdapter rowsElementsAdapter = new RowElemetsAdapter(rows.get(position).getItems().get(0),
                annotations, cardWidth, Width, context);
        RecyclerView.LayoutManager recycerows = new LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false);
        holder.rowElemetsRecycle.setLayoutManager(recycerows);
        holder.rowElemetsRecycle.setAdapter(rowsElementsAdapter);
    }

    @Override
    public int getItemCount() {
        int arr = 0;
        try{
            if(rows.size()!=0){
                arr=rows.size(); }
        }
        catch (Exception ignored){ }
        return arr;
    }

    class MyHolder extends RecyclerView.ViewHolder{
        RecyclerView rowElemetsRecycle;

        MyHolder(View itemView) {
            super(itemView);
            rowElemetsRecycle = itemView.findViewById(R.id.rowelementsrecycle);
        }
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }
}

File: app/src/main/res/layout/com/bracket/Adapters/RowsAdapter.kt
package com.bracket.Adapters

import android.annotation.SuppressLint
import android.content.Context
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bracket.Models.Annotations
import com.bracket.Models.Rows
import com.bracket.R

class RowsAdapter(private val rows: List<Rows>, private val annotations: List<Annotations>, private val cardWidth: Int, private val Width: Int, private val context: Context?) :
    RecyclerView.Adapter<RowsAdapter.MyHolder>() {

    private lateinit var rowsElementsAdapter: RowElemetsAdapter

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyHolder {

        val view = LayoutInflater.from(context).inflate(R.layout.single_row_card, parent, false)
        return MyHolder(view)
    }

    override fun onBindViewHolder(holder: MyHolder, @SuppressLint("RecyclerView") position: Int) {

        holder.rowElemetsRecycle.isNestedScrollingEnabled = false

        rowsElementsAdapter = RowElemetsAdapter(rows[position].items[0], annotations, cardWidth, Width, context)
        val recycerows = LinearLayoutManager(context, LinearLayoutManager.HORIZONTAL, false)
        holder.rowElemetsRecycle.layoutManager = recycerows
        holder.rowElemetsRecycle.adapter = rowsElementsAdapter

    }

    override fun getItemCount(): Int {
        var arr = 0
        try {
            arr = if (rows.isEmpty()) {
                0
            } else {
                rows.size
            }
        }
        catch (ignored: Exception) { }
        return arr
    }

    inner class MyHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var rowElemetsRecycle: RecyclerView = itemView.findViewById(R.id.rowelementsrecycle)

    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }
}



